airbyte-config/init/src/main/java/io/airbyte/config/init/RemoteConnectorCatalogPersistence.java

```diff
@@ -0,0 +1,225 @@
+    final HttpRequest request = HttpRequest.newBuilder(catalogUrl).timeout(timeout).header("accept", "application/json").build();
+    final HttpResponse<String> response = httpClient.send(request, HttpResponse.BodyHandlers.ofString());
+    if (response.statusCode() >= 400) {
+      throw new IOException(
+          "getRemoteConnectorCatalog request ran into status code error: " + response.statusCode() + "with message: " + response.getClass());
+    }
+    try {
+      return Jsons.deserialize(response.body());
+    } catch (final RuntimeException e) {
+      throw new IOException("Could not deserialize JSON response: ", e);
+    }
+
+  }
+
+  private ImmutableMap<AirbyteConfig, Map<String, JsonNode>> parseConnectorCatalog(JsonNode catalog) {
+    return ImmutableMap.<AirbyteConfig, Map<String, JsonNode>>builder()
+        .put(ConfigSchema.STANDARD_SOURCE_DEFINITION, parseRemoteConfigs(catalog, ConfigSchema.STANDARD_SOURCE_DEFINITION))
+        .put(ConfigSchema.STANDARD_DESTINATION_DEFINITION, parseRemoteConfigs(catalog, ConfigSchema.STANDARD_DESTINATION_DEFINITION))
+        .build();
+
+  }
+
+  private static Map<String, JsonNode> parseRemoteConfigs(final JsonNode rawConnectorCatalog, final ConfigSchema configType) {
+    final String configKey = CONFIG_SCHEMA_CONNECTOR_CATALOG_MAP.get(configType);
+    final JsonNode rawConfigs = rawConnectorCatalog.get(configKey);
+    if (rawConfigs == null) {
+      return Collections.emptyMap();
+    }
+
+    final List<JsonNode> configKeyElements = MoreIterators.toList(rawConfigs.elements());
+
+    final Map<String, JsonNode> configs = configKeyElements.stream().collect(Collectors.toMap(
+        json -> json.get(configType.getIdFieldName()).asText(),
+        json -> json));
+    final Map<String, JsonNode> configsWithMissingField = configs.entrySet().stream()
+        .collect(Collectors.toMap(Entry::getKey, e -> {
+          final JsonNode output =
+              addMissingCustomField(
+                  addMissingPublicField(
+                      addMissingTombstoneField(e.getValue())));
+          AirbyteConfigValidator.AIRBYTE_CONFIG_VALIDATOR.ensureAsRuntime(configType, output);
+          return output;
+        }));
+    return configsWithMissingField;
+  }
+
+  private static JsonNode addMissingTombstoneField(final JsonNode definitionJson) {
+    final JsonNode currTombstone = definitionJson.get("tombstone");
+    if (currTombstone == null || currTombstone.isNull()) {
+      ((ObjectNode) definitionJson).set("tombstone", BooleanNode.FALSE);
+    }
+    return definitionJson;
+  }
+
+  private static JsonNode addMissingPublicField(final JsonNode definitionJson) {
+    final JsonNode currPublic = definitionJson.get("public");
+    if (currPublic == null || currPublic.isNull()) {
+      // definitions loaded from the cloud connector catalog are by definition public
+      ((ObjectNode) definitionJson).set("public", BooleanNode.TRUE);
```

Comment:
I don't think there's anything to change on this end, but just a note that the remote catalog will now have `public` set to `true` since the schema is enforced more strictly in https://github.com/airbytehq/airbyte-cloud/pull/2485/commits/86e9dc72a5b7afd821a9f8b3eaa10986e370817f

Reply:
Ok, so if this is enforced upstream I'll remove the addition of `public` and `custom` fields.